File: Restaurant/Restaurant.Web/Modules/ServerTypes/Default/CategoryMealsColumns.ts
import { ImageViewerFormatter } from "@/Default/Product/ImageViewerFormatter";
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { CategoryMealsRow } from "./CategoryMealsRow";

export interface CategoryMealsColumns {
    Id: Column<CategoryMealsRow>;
    CategoryMealsName: Column<CategoryMealsRow>;
    Photo: Column<CategoryMealsRow>;
    Description: Column<CategoryMealsRow>;
}

export class CategoryMealsColumns extends ColumnsBase<CategoryMealsRow> {
    static readonly columnsKey = 'Default.CategoryMeals';
    static readonly Fields = fieldsProxy<CategoryMealsColumns>();
}

[ImageViewerFormatter]; // referenced types

File: Restaurant/Restaurant.Web/Modules/Default/CategoryMeals/CategoryMealsColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace Restaurant.Default.Columns;

[ColumnsScript("Default.CategoryMeals")]
[BasedOnRow(typeof(CategoryMealsRow), CheckNames = true)]
public class CategoryMealsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string CategoryMealsName { get; set; }
    [ImageViewerFormatter]
    public string Photo { get; set; }
    public string Description { get; set; }
}